mllib/src/main/scala/org/apache/spark/ml/tree/impl/DecisionTreeMetadata.scala

```diff
@@ -113,6 +113,10 @@ private[spark] object DecisionTreeMetadata extends Logging {
       throw new IllegalArgumentException(s"DecisionTree requires size of input RDD > 0, " +
         s"but was given by empty one.")
     }
+    if (numFeatures == 0) {
```

Comment:
Thank you, I've made the change to use require(numFeatures > 0, ....) in the latest commit.